ext/mysqli/tests/mysqli_real_connect.phpt

```diff
@@ -111,7 +111,11 @@ mysqli.allow_local_infile=1
 	}
 
 	mysqli_close($link);
-	@var_dump($link);
+	try {
+        @var_dump($link);
+	} catch (Error $exception) {
+	    echo $exception->getMessage() . "\n";
+	}
```

Comment:
This is  problematic. It's a pre-exiting issue that var_dump() tries to dump properties that cannot be accessed at this status, but after making things throw it's just not possible to var_dump() it at all anymore.

I think either the property accesses should be omitted from this patch, or we should add the necessary logic to not dump what can't be dumped beforehand.

Reply:
> we should add the necessary logic to not dump what can't be dumped

Does it mean that we have to inspect the passed variable recursively for possible issues with display (in order to handle `var_dump([$link])` for example)? Can it be achieved by only checking if a resource is available (although I haven't checked if it can be done in a general way) or is it something else?

Unfortunately, it seems to me that the best options is to cut the PR's scope to only deal with the functions for now (but it's enough for the mysqli stubs fortunately), and come back to the problem with `var_dump()` afterwards.

Reply:
What I had in mind here is making a change on the mysqli side. There should be a get_debug_info handler somewhere, and we can modify it to just not include the parts that aren't available in the result.

Reply:
Oh, I see now, I couldn't imagine your idea on my own. :) I'll have a look then if I am able to do it.

Reply:
Thank you, I tried something to exclude the inaccessible properties, but apparently it didn't work. So I'll continue the search tomorrow. ^^

Reply:
Now, I managed to make the tests green (just had to rebase to current master as well), although I just hope that I did the right thing in order to fix this test.